instrumentation/github.com/Shopify/sarama/otelsarama/producer.go

```diff
@@ -35,9 +35,11 @@ type syncProducer struct {
 
 // SendMessage calls sarama.SyncProducer.SendMessage and traces the request.
 func (p *syncProducer) SendMessage(msg *sarama.ProducerMessage) (partition int32, offset int64, err error) {
-	span := startProducerSpan(p.cfg, p.saramaConfig.Version, msg)
+	span, ok := startProducerSpan(p.cfg, p.saramaConfig.Version, msg)
 	partition, offset, err = p.SyncProducer.SendMessage(msg)
-	finishProducerSpan(span, partition, offset, err)
+	if ok {
+		finishProducerSpan(span, partition, offset, err)
+	}
```

Comment:
this design is very error-prone. why not moving the `if` logic to `finishProducerSpan`